MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -149,12 +149,26 @@ class MSALNativeAuthBaseController {
         completion(response, error)
     }
 
-    func performRequest<T>(_ request: MSIDHttpRequest, context: MSIDRequestContext) async -> Result<T, Error> {
+    func performRequest<T: MSALNativeAuthResponseCorrelatable>(
+        _ request: MSIDHttpRequest,
+        context: MSALNativeAuthRequestContext
+    ) async -> Result<T, Error> {
         return await withCheckedContinuation { continuation in
-            request.send { result, error in
+            request.send { [weak self] result, error in
                 if let error = error {
+                    if let correlationId = self?.extractCorrelationIdFromUserInfo((error as NSError).userInfo) {
+                        context.setServerCorrelationId(UUID(uuidString: correlationId))
+                    } else if let errorWithCorrelationId = error as? MSALNativeAuthResponseCorrelatable {
+                        context.setServerCorrelationId(errorWithCorrelationId.correlationId)
+                    } else if case MSALNativeAuthInternalError.responseSerializationError(let correlationId) = error {
+                        context.setServerCorrelationId(correlationId)
+                    } else {
+                        MSALLogger.log(level: .warning, context: context, format: "Error request - cannot decode error headers. Continuing")
```

Comment:
I think we should set serverCorrelation Id to nil here. If don't do it, the serverCorrelationId would be still the one from the previous response, so once we retrieve a correlation Id from the context, we will use the one from the wrong one

Reply:
Nice one, thanks.